cristin-organization/src/main/java/no/unit/nva/cristin/organization/CristinApiClient.java

```diff
@@ -45,15 +52,74 @@ public Organization getOrganization(URI uri)
      *
      * @param requestQueryParams Map containing verified query parameters
      */
-    public SearchResponse<Organization> queryInstitutions(Map<String, String> requestQueryParams) {
-        return new SearchResponse<Organization>(
-                new UriWrapper(HTTPS, DOMAIN_NAME)
-                        .addChild(BASE_PATH)
-                        .addChild(UriUtils.INSTITUTION)
-                        .addQueryParameters(requestQueryParams)
-                        .getUri())
-                .withProcessingTime(0L)
-                .withSize(0)
-                .withHits(Collections.emptyList());
+    public SearchResponse<Organization> queryOrganizations(Map<String, String> requestQueryParams)
+            throws NotFoundException, FailedHttpRequestException {
+        Map<String, String> cristinRequestQueryParams = translateToCristinApi(requestQueryParams);
+        URI queryUri = new UriWrapper(HTTPS, CRISTIN_API_BASE)
+                .addChild(UNITS_PATH)
+                .addQueryParameters(cristinRequestQueryParams)
+                .getUri();
+
+        long start = System.currentTimeMillis();
+        SearchResponse<Organization> searchResponse = httpExecutor.query(queryUri);
+        final long totalProcessingTime = System.currentTimeMillis() - start;
+        return updateSearchResponseMetadata(searchResponse,requestQueryParams, totalProcessingTime);
+    }
+
+    private Map<String, String> translateToCristinApi(Map<String, String> requestQueryParams) {
+        return Map.of(
+                CRISTIN_QUERY_NAME_PARAM, requestQueryParams.get(QUERY),
+                PAGE, requestQueryParams.get(PAGE),
+                CRISTIN_PER_PAGE_PARAM, requestQueryParams.get(NUMBER_OF_RESULTS));
+    }
+
+    private SearchResponse<Organization> updateSearchResponseMetadata(
+            SearchResponse<Organization> searchResponse,
+            Map<String, String> requestQueryParams,
+            long timeUsed) {
+        searchResponse.setId(getNvaQueryId(requestQueryParams));
+        searchResponse.setFirstRecord(calculateFirstRecord(requestQueryParams));
+        searchResponse.setNextResults(nextResult(getNvaApiBaseUri(), requestQueryParams, searchResponse.getSize()));
+        searchResponse.setPreviousResults(previousResult(getNvaApiBaseUri(),
+                requestQueryParams,
+                searchResponse.getSize()));
+        searchResponse.setProcessingTime(timeUsed);
+        return searchResponse;
+    }
+
+    private URI getNvaQueryId(Map<String, String> requestQueryParams) {
+        return new UriWrapper(getNvaApiBaseUri()).addQueryParameters(requestQueryParams).getUri();
+    }
+
+    private URI getNvaApiBaseUri() {
+        return new UriWrapper(HTTPS, DOMAIN_NAME).addChild(BASE_PATH).addChild(ORGANIZATION_PATH).getUri();
+    }
+
+    private URI previousResult(URI baseUri, Map<String, String> requestQueryParams, int totalSize) {
```

Comment:
Skeptical about duplicating the code when there is something similar in SearchResponse, but that can be refactored later.

Reply:
It creates uri against cristin api, not nva - we have to make it neutral to the uri